### PR TITLE
Always look for adapters in Test Source directory

### PR DESCRIPTION
**Pull request by mayankbansal018**
## Description

- Allow Testadapters to be picked from source dir for fullCLR projects
- Allow dotnet CLI to take multiple inputs for logger, & TAP

Dotnet CLI PR: https://github.com/dotnet/cli/pull/9198

## Related issue
Fixes https://github.com/Microsoft/vstest/issues/1319

### CODE REVIEW

File: test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Hosting/DefaultTestHostManagerTests.cs [hunk 1]
```diff
@@ -219,8 +219,11 @@ public void GetTestPlatformExtensionsShouldExcludeOutputDirectoryExtensionsIfTes
             List<string> extensionsList1 = new List<string> { @"C:\Source1\ext1.TestAdapter.dll", @"C:\Source1\ext2.TestAdapter.dll" };
             this.mockFileHelper.Setup(fh => fh.EnumerateFiles(sourcesDir[0], SearchOption.TopDirectoryOnly, "TestAdapter.dll")).Returns(extensionsList1);
 
+            this.mockFileHelper.Setup(fh => fh.GetFileVersion(extensionsList1[0])).Returns(new Version(2, 0));
+            this.mockFileHelper.Setup(fh => fh.GetFileVersion(extensionsList1[1])).Returns(new Version(5, 5));
+
             this.testHostManager.Initialize(this.mockMessageLogger.Object, $"<?xml version=\"1.0\" encoding=\"utf-8\"?><RunSettings> <RunConfiguration><TestAdaptersPaths>C:\\Foo</TestAdaptersPaths></RunConfiguration> </RunSettings>");
-            List<string> currentList = new List<string> { @"FooExtension.dll" };
+            List<string> currentList = new List<string> { @"FooExtension.dll", @"C:\Source1\ext1.TestAdapter.dll", @"C:\Source1\ext2.TestAdapter.dll" };
```
Comment (smadala) [inline test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Hosting/DefaultTestHostManagerTests.cs]: Change test name.

File: src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs [hunk 2]
```diff
@@ -192,7 +191,7 @@ public TestHostConnectionInfo GetTestHostConnectionInfo()
         /// <inheritdoc/>
         public IEnumerable<string> GetTestPlatformExtensions(IEnumerable<string> sources, IEnumerable<string> extensions)
         {
-            if (sources != null && sources.Any() && this.projectOutputExtensionsRequired)
+            if (sources != null && sources.Any())
```
Comment (smadala) [inline src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs]: Remove explicitly testadapterpath from acceptance tests. [Example.](https://github.com/Microsoft/vstest/blob/master/test/Microsoft.TestPlatform.AcceptanceTests/DataCollectionTests.cs#L52)

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs [hunk 3]
```diff
@@ -139,9 +140,12 @@ internal IEnumerable<string> CreateArgument()
...
-            if (!string.IsNullOrEmpty(this.VSTestTestAdapterPath))
+            if (this.VSTestTestAdapterPath != null && this.VSTestTestAdapterPath.Length > 0)
             {
-                allArgs.Add("--testAdapterPath:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(this.VSTestTestAdapterPath));
+                foreach (var arg in this.VSTestTestAdapterPath)
+                {
+                    allArgs.Add("--testAdapterPath:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(arg));
+                }
             }
             else
```
Comment (smadala) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs]: We can remove else condition.

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs [hunk 4]
```diff
@@ -168,9 +172,17 @@ internal IEnumerable<string> CreateArgument()
...
+            if (this.VSTestLogger != null && this.VSTestLogger.Length > 0)
             {
-                allArgs.Add("--logger:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(this.VSTestLogger));
+                foreach (var arg in this.VSTestLogger)
+                {
+                    allArgs.Add("--logger:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(arg));
+
+                    if (arg.StartsWith("console", StringComparison.OrdinalIgnoreCase))
+                    {
+                        initializeConsoleLogger = false;
```
Comment (smadala) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs]: Nit: Rename to `isConsoleLoggerEnabled` or something close to it?

File: test/Microsoft.TestPlatform.Build.UnitTests/VsTestTaskTests.cs [hunk 5]
```diff
@@ -234,5 +234,40 @@ public void CreateArgumentShouldAddOneCollectArgumentForEachCollect()
...
+            // Add values for required properties.
+            vstestTask.TestFileFullPath = "abc";
+            vstestTask.VSTestFramework = "abc";
+
+            vstestTask.VSTestTestAdapterPath = new string[] { "path1", "path2" };
+
+            var allArguments = vstestTask.CreateArgument().ToArray();
+
+            Assert.IsNotNull(allArguments.FirstOrDefault(arg => arg.Contains("--testAdapterPath:path1")));
+            Assert.IsNotNull(allArguments.FirstOrDefault(arg => arg.Contains("--testAdapterPath:path1")));
```
Comment (smadala) [inline test/Microsoft.TestPlatform.Build.UnitTests/VsTestTaskTests.cs]: `path2`.

File: test/Microsoft.TestPlatform.Build.UnitTests/VsTestTaskTests.cs [hunk 6]
```diff
@@ -234,5 +234,40 @@ public void CreateArgumentShouldAddOneCollectArgumentForEachCollect()
...
+        public void CreateArgumentShouldAddMultipleLoggers()
+        {
+            var vstestTask = new VSTestTask();
+
+            // Add values for required properties.
+            vstestTask.TestFileFullPath = "abc";
+            vstestTask.VSTestFramework = "abc";
+
+            vstestTask.VSTestLogger = new string[] { "trx;LogFileName=foo bar.trx", "console" };
+
```
Comment (singhsarab) [inline test/Microsoft.TestPlatform.Build.UnitTests/VsTestTaskTests.cs]: Nit: extra line
Also, better for required properties, use new VSTestTask { TestFileFullPath = "abc", VSTestFramework = "xyz"}

**Review by singhsarab — APPROVED** [review]
Please mention the testing done with respect to changes associated with dotnet cli change.
Approving with suggestions.

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs [hunk 7]
```diff
@@ -31,7 +31,7 @@ public string VSTestSetting
             set;
         }
 
-        public string VSTestTestAdapterPath
+        public string[] VSTestTestAdapterPath
```
Comment (cltshivash) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs]: >string[] [](start = 15, length = 8)

nitpick: IEnumerable over string[]

**Comment by cltshivash** [comment]
>     public class VSTestTask : Task, ICancelableTask

is this shipping code ?

---
Refers to: src/Microsoft.TestPlatform.Build/Tasks/VSTestTask.cs:15 in e6db9d0. [](commit_id = e6db9d00b998b596ba95826259475f642df4aeb5, deletion_comment = False)

**Review by smadala — APPROVED** [review]

**Comment by mayankbansal018** [comment]
@dotnet-bot test Windows_NT / Debug Build please

**Comment by mayankbansal018** [comment]
@dotnet-bot test Windows_NT / Debug Build please

**Pull request merged**